src/models.py

```diff
@@ -0,0 +1,178 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class Deshadower(nn.Module):
+    def __init__(self, in_channels, out_channels, res_blocks=9):
+        super(Deshadower, self).__init__()
+
+        # Initial conv layer
+        self.model = nn.Sequential(
+            nn.ReflectionPad2d(3),
+            nn.Conv2d(in_channels, 64, 7),
+            nn.InstanceNorm2d(64),
+            nn.ReLU(inplace=True),
+        )
+
+        # downsampling
+        in_channels = 64
+        out_channels = in_channels * 2
+
+        downsampling = (
+            nn.Conv2d(in_channels, out_channels, kernel_size=3, stride=2, padding=1),
+            nn.InstanceNorm2d(out_channels),
+            nn.ReLU(inplace=True),
+        )
+
+        for _ in range(2):
```

Comment:
hardcoded

Reply:
So you suggest creating an argument in Deshadower's constructor to manipulate downsampling iterations?

Reply:
We can try that, however in the for loop in_channels as well as out_channels are changing. If we allow more (or less) iterations than 2, it may affect residual blocks and the rest of the layers.

Reply:
set it as seperate variable eg downsampling_iterations or as a paremeter with default value